### PR TITLE
unsubscribe page

### PR DESCRIPTION
**Pull request by emnlpstr**
new page unsubscribe

## Checklist

- [x] I followed the indications in the [CONTRIBUTING](CONTRIBUTING.md)
- [x] The documentation related to the proposed change has been updated accordingly (also comments in code).
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?
- [x] Ready for review! :rocket:

### CODE REVIEW

**Review by lollomag — APPROVED** [review]

File: src/layouts/UnsubscribePage.jsx [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  return state;
+};
+
+const initState = {
+  status: LOADING,
+};
+
+export const UnsubscribePage = ({ location }) => {
+  const params = new URLSearchParams(location.search);
+  const jwt = params.get('jwt');
```
Comment (sebbalex) [inline src/layouts/UnsubscribePage.jsx:48]:
```suggestion
  const address = params.get('address');
  const uuid = params.get('uuid');
```

File: src/layouts/UnsubscribePage.jsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  const [state, dispatch] = useReducer(reducer, initState);
+
+  const unsubscribe = useCallback(async () => {
+    if (jwt && !jwt.match(/(^[\w-]*\.[\w-]*\.[\w-]*$)/)) {
+      dispatch({
+        type: ERROR,
+        payload: `jwt not present or it seems it is not a valid jwt, ${jwt}`,
+      });
+      return;
+    }
```
Comment (sebbalex) [inline src/layouts/UnsubscribePage.jsx:63]:
```suggestion
```

File: src/layouts/UnsubscribePage.jsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    const { address, uuid } = jwt_decode(jwt);
```
Comment (sebbalex) [inline src/layouts/UnsubscribePage.jsx:64]:
```suggestion
```

File: src/layouts/UnsubscribePage.jsx [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    const options = {
+      crossDomain: true,
+      method: 'PATCH',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ jwt }),
```
Comment (sebbalex) [inline src/layouts/UnsubscribePage.jsx:71]:
```suggestion
```

File: src/layouts/UnsubscribePage.jsx [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+      const message = await response.json();
+      if (status >= 200 && status <= 299) {
+        dispatch({ type: SUCCESS, payload: { message, status } });
+      } else {
+        dispatch({ type: ERROR, payload: { message, status } });
+      }
+    } catch (error) {
+      dispatch({ type: ERROR, payload: error });
+    }
+  }, [apiUrl, jwt]);
```
Comment (sebbalex) [inline src/layouts/UnsubscribePage.jsx:85]:
```suggestion
  }, [apiUrl]);
```

**Review by sebbalex — CHANGES_REQUESTED** [review]
`address` and `uuid` are search parameters, we don't have `jwt` here.

**Review by sebbalex — APPROVED** [review]

**Pull request merged**